pkg/apis/core/v1alpha1/deployer_lifecycle.go

```diff
@@ -89,6 +89,10 @@ func (ds *DeployerStatus) PropagateServiceStatus(ss *corev1.ServiceStatus) {
 	deployerCondSet.Manage(ds).MarkTrue(DeployerConditionServiceReady)
 }
 
+func (ds *DeployerStatus) MarkServiceReconcileError(err error) {
```

Comment:
I'll adopt this. I overlooked that method.

Reply:
Actually, I see that riff 0.4 was checking services ahead of reconciliation, which is a design we've moved away from. Rather than trying to introspect the service creation error, which is fragile, I think it's sufficient to capture the service creation error in the service ready condition.